### PR TITLE
Fix some typos in documentation, code, code comments etc.

### PR DESCRIPTION
**Pull request by stweil**
Most of them were found using codespell and typos.

### CODE REVIEW

File: parser/parser_test.go [hunk 1]
```diff
@@ -536,7 +536,7 @@ PARAMETER param1 value1
 PARAMETER param2 value2
 TEMPLATE template1
 MESSAGE system """
-You are a store greeter. Always responsed with "Hello!".
+You are a store greeter. Always responded with "Hello!".
```
Comment (stweil) [inline parser/parser_test.go:539]: I am not sure about this change. Maybe there is a better fix for this line and also for line 557 below.

Comment (jmorganca) [inline parser/parser_test.go:539]:
```suggestion
You are a store greeter. Always respond with "Hello!".
```

File: parser/parser_test.go [hunk 2]
```diff
@@ -554,7 +554,7 @@ PARAMETER param1 value1
...
```
Comment (jmorganca) [inline parser/parser_test.go:557]:
```suggestion
You are a store greeter. Always respond with "Hello!".
```

**Review by jmorganca — APPROVED** [review]
Small comment otherwise LGTM. Thanks!

**Comment by stweil** [comment]
> Small comment otherwise LGTM. Thanks!

Thanks for the suggested changes which were now applied.

File: server/testdata/tools/mistral.out [hunk 3]
```diff
@@ -1,3 +1,3 @@
...
+[INST] What's the weather like today in Paris?[/INST][TOOL_CALLS] [{"name": "get_current_weather", "arguments": {"format":"celsius","location":"Paris, France"}}]</s>[TOOL_RESULTS] {"content": 22}[/TOOL_RESULTS] The current temperature in Paris, France is 22 degrees Celsius.</s>[AVAILABLE_TOOLS] [{"type":"function","function":{"name":"get_current_weather","description":"Get the current weather","parameters":{"type":"object","required":["location","format"],"properties":{"format":{"type":"string","description":"The temperature unit to use. Infer this from the user's location.","enum":["celsius","fahrenheit"]},"location":{"type":"string","description":"The city and state, e.g. San Francisco, CA"}}}}}][/AVAILABLE_TOOLS][INST] You are a knowledgeable assistant. You can answer questions and perform tasks.
 
-What's the weather like today in San Francisco and Toronto?[/INST]
\ No newline at end of file
+What's the weather like today in San Francisco and Toronto?[/INST]
```
Comment (stweil) [inline server/testdata/tools/mistral.out:3]: @jmorganca, it looks like the linefeeds at the file endings which were added by my editor (vi) cause a failure of the CI tests.

There are two possible solutions:
1. I could remove the added linefeed which makes the tests fail.
2. I could update the tests.

Solution 1 would restore the old state without a terminating linefeed. This is rather unusual for text files, but maybe desired in this project?

Which solution is preferred?

**Comment by dhiltgen** [comment]
Thanks for taking the time to post a PR. I noticed you've made some changes to server.cpp, so I wanted to let you know that we're about to merge another PR (#5034) to begin replacing that code with a new Go based equivalent. The goal of this is to add more unit testing and fix some long standing stability bugs while preserving vision model support. If you need help rebasing your PR don't hesitate to contact us by replying here.

**Comment by stweil** [comment]
@dhiltgen, , the pull request is rebased and the conflicts are fixed.

**Comment by jmorganca** [comment]
Hi @stweil so very sorry – would it be possible to rebase again?

**Comment by jmorganca** [comment]
Would it be possible to avoid changing files in the `llama/` directory for this PR – as it's changing vendor code.

**Comment by stweil** [comment]
> Would it be possible to avoid changing files in the `llama/` directory for this PR – as it's changing vendor code?

Sure, I removed those changes now.

**Review by jmorganca — APPROVED** [review]

**Pull request merged**